lib/cocoapods-core/specification/dsl.rb

```diff
@@ -676,6 +676,11 @@ def dependency(*args)
             end
           end
         end
+
+        options = version_requirements.last.is_a?(Hash) ? version_requirements.pop.dup : {}
+        whitelist = options.delete(:configurations)
+        raise Informative, "Unsupported options: #{options}" unless options.empty?
```

Comment:
needs to be merged with the hash-handling logic below

Reply:
may have a fix for this.